### DISCUSSION TITLE
Incomplete downloads

### DISCUSSION THREAD

**Discussion by abeych** [curl/curl#6706] [Q&A]
I have seen a scenario where there was an incomplete download happening. The curl_easy_perform() function returned before completing the download and returned CURL_OK. The total downloaded bytes was less than the total size of file to be downloaded. Signature check too failed because of this. Is this a known issue? Was there any fix given for this in any releases?

**Comment by jay** [comment]
Try the [latest version of curl](https://curl.se/download.html) and if it's still a problem please [open an issue](https://github.com/curl/curl/issues/new/choose) and give us a way to reproduce.

**Reply by jay** [reply]
If the latest release version doesn't work then check to see if you are affected by #6736.
